// @aws-sdk/client-sagemaker/dist-cjs/models/errors.js
const { SageMakerServiceException: __BaseException } = require("./SageMakerServiceException");
exports.ResourceLimitExceeded = class ResourceLimitExceeded extends __BaseException {
    name = "ResourceLimitExceeded";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ResourceLimitExceeded",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceLimitExceeded.prototype);
        this.Message = opts.Message;
    }
};
exports.ResourceNotFound = class ResourceNotFound extends __BaseException {
    name = "ResourceNotFound";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ResourceNotFound",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceNotFound.prototype);
        this.Message = opts.Message;
    }
};
exports.ConflictException = class ConflictException extends __BaseException {
    name = "ConflictException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ConflictException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConflictException.prototype);
        this.Message = opts.Message;
    }
};
exports.ResourceInUse = class ResourceInUse extends __BaseException {
    name = "ResourceInUse";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ResourceInUse",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceInUse.prototype);
        this.Message = opts.Message;
    }
};
